src/CopyrighFixer/CopyrighFixer/signer.cpp

```diff
@@ -5,27 +5,135 @@
 //# of this license document, but changing it is not allowed.
 //#
 
+#include <time.h>
 #include "signer.h"
 #include "CopyrighFixer/ifilemanager.h"
+#include <QDir>
+#include <quasarapp.h>
 
 namespace CopyrighFixer {
 Signer::Signer() {
 
 }
 
+
+bool Signer::processSign(const QString &pathToFile, const Config &objConf) const {
+
+    QDir currentFolder(pathToFile);
+    currentFolder.setFilter(QDir::Dirs | QDir::Files | QDir::NoDotAndDotDot);
+
+    QFileInfoList folderItems(currentFolder.entryInfoList());
+
+    for (auto i_file: folderItems) {
+
+        if (i_file.isDir()) {
+            processSign(i_file.filePath(), objConf);
+        }
+
+        IFileManager *currFM = searchFileByExt(i_file.suffix());
+        Signature signFromFile;
+        Signature resultSign;
+
+        if (currFM == nullptr) {
+            return false;
```

Comment:
Please add Error message wit description of this error. 
example "The CFixer tool are not support  files with sufix: .sufix"